packages/flutter/lib/src/services/text_input.dart

```diff
@@ -671,6 +678,87 @@ class RawFloatingCursorPoint {
   final FloatingCursorDragState state;
 }
 
+/// A [TextEditingDeltaType.insertion] singifies there has been a single/sequence
+/// character insertion.
+/// A [TextEditingDeltaType.replacement] signifies that a replacement has occured.
+/// A replacement can occur in cases such as auto-correct, suggestions, and
+/// when a selection is replaced by a single character.
+/// A [TextEditingDeltaType.deletion] signifies there has been a single/sequence
+/// character deletion.
+/// A [TextEditingDeltaType.equality] signifies no text has changed from the last
+/// [TextEditingValue].
+enum TextEditingDeltaType { insertion, deletion, replacement, equality }
+
+/// A model representing the changes made to a [TextEditingValue] from time T1
+/// to T2.
+class TextEditingDelta {
+  const TextEditingDelta({
+    required this.oldText,
+    required this.deltaText,
+    required this.deltaType,
+    required this.deltaRange,
+    required this.selection,
+    required this.composing,
+  });
+
+  /// The old text state before the delta has occured.
+  final String oldText;
```

Comment:
We might actually be able to remove oldText all together what do you think? The framework technically has the "oldText" already so I don't think we need to send it.

Reply:
It feels a little bit weird to send a delta without any indication of what it applies to, but I guess it makes sense to remove it for now since we currently haven't seen a need for it.  Keep an eye out for bugs with the engine and framework getting out of sync though.  If that seems possible then maybe we can add a hash in the future.

Reply:
Update: Seems like we're keeping it in order to make sure the delta applies to the current text, for now, but we could investigate removing it for space savings later.